src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -854,7 +869,8 @@ public static IEnumerable<TSource> Take<TSource>(this IEnumerable<TSource> sourc
             if (count <= 0) return new EmptyPartition<TSource>();
             IPartition<TSource> partition = source as IPartition<TSource>;
             if (partition != null) return partition.Take(count);
-            return TakeIterator<TSource>(source, count);
+            IList<TSource> sourceList = source as IList<TSource>;
+            return (sourceList != null) ? new SkipListIterator<TSource>(sourceList, 0, count - 1) : TakeIterator(source, count);
```

Comment:
Probably a better name. The current one shows its legacy more than its function.